codespell_lib/tests/test_basic.py

```diff
@@ -64,13 +64,15 @@ def main(
 
 
 def run_codespell(
-    args: Tuple[Any, ...] = (),
-    cwd: Optional[Path] = None,
+    args: Tuple[Any, ...],
+    cwd: Optional[Path],
 ) -> int:
     """Run codespell."""
+    lib = "codespell_lib"
+    copytree(lib, cwd / lib, dirs_exist_ok=True)
```

Comment:
Where exactly does this copy `codespell_lib`? Wherever the user is, for example in `codespell_lib/tests`? Can't we avoid a copy?

I don't see [flit/tests/test_command.py](https://github.com/pypa/flit/tree/main) do that.

Reply:
When running the `python -m codespell_lib` command, the codespell script runs from the current directory.  This means that it *may* parse all the files in the project.

The solution is to set `cwd` to a temporary directory, but in this case you need to have `codespell_lib` in the new current directory.

IMHO a copy cannot be avoided, but we can ignore the `data` and `tests` directory using `shutil.ignore_patterns`.